core/src/main/scala/org/apache/spark/util/LocalDirectoryCleanupUtil.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright (c) 2018 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package org.apache.spark.util
+
+import java.io.File
+import java.nio.file.{Files, Path, Paths}
+import java.util.Collections
+import java.util.function.Consumer
+
+import scala.io.Source
+
+import com.google.common.collect.Lists
+import org.apache.commons.io.FileUtils
+
+import org.apache.spark.internal.Logging
+
+/**
+ * Contains utility methods for cleaning of spark local directories left orphan due to scenario
+ * like abrupt failure of JVM.
+ */
+object LocalDirectoryCleanupUtil extends Logging {
+
+  private lazy val listFile = ".tempfiles.list"
+
+  /**
+   * Add new path to temporary file list.
+   *
+   * @param file temp file/directory
+   */
+  def add(file: File): Unit = {
+    FileUtils.writeLines(new File(listFile), "UTF-8",
+      Lists.newArrayList(file), true)
+  }
+
+  /**
+   * Attempts to recursively delete all files/directories present in temp files list.
+   * Also cleans the temp files list once deletion is complete.
+   */
+  def clean(): Unit = {
+    val listFilePath = Paths.get(listFile)
+    if (Files.exists(listFilePath)) {
+      val fileSource = Source.fromFile(listFile, "UTF-8")
+      try {
+        fileSource.getLines().map(Paths.get(_)).foreach(delete)
+      } finally {
+        fileSource.close()
+      }
+      try {
+        Files.delete(listFilePath)
+      } catch {
+        case ex: Exception => logError(s"Failure while deleting file: $listFile.", ex)
+          System.exit(1)
```

Comment:
Better to use ExecutorExitCode.DISK_STORE_FAILED_TO_CREATE_DIR. Then ExecutorExitCode.explainExitCode the message can be changed to "Failed to create or delete local ...".

Reply:
@sumwale At this point, we are deleting `.tempfiles.list` which is the metadata file used to track the orphan spark local dirs. Actual spark temp directories and files contained in that directory is deleted as part of  `org.apache.spark.util.LocalDirectoryCleanupUtil#delete` method. The message in `ExecutorExitCode.explainExitCode` is more specific to `spark.local.dir`. Is this suggestion actually targeted to deletiong of `spark.local.dir` and it's content?

Reply:
Ok, current one should be ok